tests/csv_test.py

```diff
@@ -0,0 +1,132 @@
+import hdat.hdat_cli as hdat
+
+
+class TestGetKeys:
+    def test_keys_default(self, mock_results, mock_keys):
+        assert hdat.get_result_keys(mock_results[0], mock_keys) == (
+            ['case_id', 'ran_on', 'result_id']
+        )
+
+    def test_specific_metrics(self, mock_results):
+        in_keys = ['case_id', 'result_id', 'metrics.mean', 'metrics.std']
+
+        assert hdat.get_result_keys(mock_results[0], in_keys) == (
+            ['case_id', 'result_id']

```

Comment:
Should `mean` and `std` be in this list?

Reply:
OK, that makes sense. I don't see successful specific dot selections without wildcards being tested -- basically the mock `test_with_metrics` test with just one or two `metrics` selections would test this out.

Reply:
Tried this out on the command line:
```
 hdat csv example-suite --keys "case_id, metrics.max"
Keys not found:  metrics.max
case_id
wheat
chicken
```

Reply:
try `case_id,metrics.max` without quotes

Reply:
@dillonwilliams 

Fixed it. Turns out, I had forgotten to implement a direct checker for nested dictionaries.

I also added in a whitespace remover because it was searching for ` metrics.max` instead of `metrics.max` due to the space after the comma.